thoth/storages/graph/postgres.py

```diff
@@ -4822,6 +4894,57 @@ def sync_revsolver_result(self, document: Dict[str, Any]) -> None:
                     platform=entry["platform"],
                 )
 
+    def _check_package_solved(self, session: Session, package_name: str, package_version: str,
+                              package_index: str, os_name: str, os_version: str, python_version: str) -> Tuple:
+        """We check if the package has been solved before syncing it to the database."""
+        # Check if the package has a known index
+        if package_index:
+            python_package_version = (
+                session.query(PythonPackageVersion)
+                .join(PythonPackageIndex)
+                .filter(PythonPackageVersion.package_name == package_name)
+                .filter(PythonPackageVersion.package_version == package_version)
+                .filter(PythonPackageVersion.os_name == os_name)
+                .filter(PythonPackageVersion.os_version == os_version)
+                .filter(PythonPackageVersion.python_version == python_version)
+                .filter(PythonPackageIndex.url == package_index if package_index else None)
+                .first()
+            )
+        else:
+            raise PythonIndexNotProvided(
+                f"Trying to sync package {package_name!r} in version {package_version!r} "
+                "which does not have corresponding Python entity record"
+            )
+        # Check if we run the solver for a specific package already
+        if python_package_version:
+            python_package_version = self._create_python_package_version(
+                session,
+                package_name=package_name,
+                package_version=package_version,
+                os_name=os_name,
+                os_version=os_version,
+                python_version=python_version,
+                index_url=package_index,
+                sync_only_entity=False,
+            )
+        else:
+            raise SolverNotRun(
+                f"Trying to sync package {package_name!r} in version {package_version!r} "
+                f"not solved by solver-{os_name}-{os_version}-{python_version}"
+            )
+        # We return python_package_entity id and python_package_version id if solver has been run. 
+        python_package_index = self._get_or_create_python_package_index(
+                    session, index_url=package_index, only_if_enabled=True
+                )
+        python_package_version_entity, _ = PythonPackageVersionEntity.get_or_create(
+            session,
+            package_name=package_name,
+            package_version=package_version,
+            python_package_index_id=python_package_index.id,
+        )
```

Comment:
Not required, we don't create anything in SI sync beside record in SI table and relation table :)